developerportal/apps/articles/models.py

```diff
@@ -102,12 +113,27 @@ def can_create_at(cls, parent):
     def get_context(self, request):
         context = super().get_context(request)
         context["filters"] = self.get_filters()
+        context["resources"] = self.get_resources(request)
         return context
 
-    @property
-    def resources(self):
+    def get_resources(self, request):
         # This Page class will show both Articles/Posts and Videos in its listing
-        return get_combined_articles_and_videos(self)
+
+        # We can't use __in in this deeply related query, so we have to make
+        # a custom Q object instead and pass is in as a filter, then deal with
+        # it later
```

Comment:
I don't understand why `__in` doesn't work here. In Kuma I tested `SocialAccount.objects.filter(user__email__in=['gregorwbr@gmail.com', 'gweber@mozilla.com'])` and it seemed to do the right thing. Does it stop working once the relation is not just between two but three models?

Reply:
Good question: I tried __in originally, too, but coudn't get it to work, instead triggering a lookup failure. I think there's a limitation going on because of limitations on the `ParentalKey` in the chain of related objects (that non-Django field class comes from `django-modelcluster`). I think the `__` relations can walk more than three in Django, just not when there's a Wagtail `ParentalKey` involved

Reply:
For topics it does seem to work and do the same thing with:
```python
q_object = Q(topics__topic__slug__in=topics) if topics else Q()
```
We might want to investigate switching to this if it works for the other cases as well. What do you think?

Reply:
@Gregoor Switched to this approach -- thanks for the poke!

Could i get a last look and hopefully a ✅ ?